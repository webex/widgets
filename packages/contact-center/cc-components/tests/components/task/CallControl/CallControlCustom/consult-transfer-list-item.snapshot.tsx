import React from 'react';
import '@testing-library/jest-dom';
import {render, fireEvent, act} from '@testing-library/react';
import ConsultTransferListComponent from '../../../../../src/components/task/CallControl/CallControlCustom/consult-transfer-list-item';

const mockUIDProps = (container) => {
  container
    .querySelectorAll('[id^="mdc-input"]')
    .forEach((el: HTMLBaseElement) => el.setAttribute('id', 'mock-input-id'));
  container
    .querySelectorAll('[id^="mdc-tooltip"]')
    .forEach((el: HTMLBaseElement) => el.setAttribute('id', 'mock-tooltip-id'));
  container
    .querySelectorAll('[aria-describedby^="mdc-tooltip"]')
    .forEach((el: HTMLBaseElement) => el.setAttribute('aria-describedby', 'mock-aria-describedby'));
};

// Mock lottie-web
jest.mock('lottie-web');

describe('ConsultTransferListComponent Snapshots', () => {
  const mockLogger = {
    log: jest.fn(),
    info: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
    debug: jest.fn(),
    trace: jest.fn(),
  };

  const mockOnButtonPress = jest.fn();

  const defaultProps = {
    title: 'John Doe',
    subtitle: 'Manager',
    buttonIcon: 'test-icon',
    onButtonPress: mockOnButtonPress,
    className: 'custom-class',
    logger: mockLogger,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component', () => {
    it('should render the component with title and subtitle', async () => {
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...defaultProps} />);
      });

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should render with long name generating different initials', async () => {
      const longNameProps = {...defaultProps, title: 'Alexander Benjamin Christopher'};
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...longNameProps} />);
      });

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should render with single name', async () => {
      const singleNameProps = {...defaultProps, title: 'John'};
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...singleNameProps} />);
      });

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should render without subtitle', async () => {
      const noSubtitleProps = {...defaultProps, subtitle: ''};
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...noSubtitleProps} />);
      });

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should render with different className', async () => {
      const customClassProps = {...defaultProps, className: 'different-custom-class'};
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...customClassProps} />);
      });

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should render with different button icon', async () => {
      const differentIconProps = {...defaultProps, buttonIcon: 'different-icon'};
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...differentIconProps} />);
      });

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should render without className', async () => {
      const noClassProps = {...defaultProps, className: undefined};
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...noClassProps} />);
      });

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should render with special characters in title', async () => {
      const specialCharsProps = {...defaultProps, title: "John O'Connor & Jane"};
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...specialCharsProps} />);
      });

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should render with long subtitle', async () => {
      const longSubtitleProps = {
        ...defaultProps,
        subtitle: 'Senior Manager of Customer Relations and Business Development',
      };
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...longSubtitleProps} />);
      });

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });
  });

  describe('Interactions', () => {
    it('should render component after button click', async () => {
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...defaultProps} />);
      });

      const button = screen.container.querySelector('button');
      if (button) {
        fireEvent.click(button);
      }

      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should render component after list item click', async () => {
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...defaultProps} />);
      });

      const listItem = screen.container.querySelector('.call-control-list-item');
      fireEvent.click(listItem);

      mockUIDProps(listItem);
      expect(listItem).toMatchSnapshot();
    });

    it('should render component after keyboard interaction', async () => {
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...defaultProps} />);
      });

      const listItem = screen.container.querySelector('.call-control-list-item');
      fireEvent.keyDown(listItem, {key: 'Enter', code: 'Enter'});

      mockUIDProps(listItem);
      expect(listItem).toMatchSnapshot();
    });
  });

  describe('State Management', () => {
    it('should update when title changes', async () => {
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...defaultProps} />);
      });

      screen.rerender(<ConsultTransferListComponent {...defaultProps} title="Jane Smith" />);
      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should update when subtitle changes', async () => {
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...defaultProps} />);
      });

      screen.rerender(<ConsultTransferListComponent {...defaultProps} subtitle="Senior Manager" />);
      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });

    it('should update when buttonIcon changes', async () => {
      let screen;
      await act(async () => {
        screen = render(<ConsultTransferListComponent {...defaultProps} />);
      });

      screen.rerender(<ConsultTransferListComponent {...defaultProps} buttonIcon="new-icon" />);
      const container = screen.container.querySelector('.call-control-list-item');
      mockUIDProps(container);
      expect(container).toMatchSnapshot();
    });
  });
});
