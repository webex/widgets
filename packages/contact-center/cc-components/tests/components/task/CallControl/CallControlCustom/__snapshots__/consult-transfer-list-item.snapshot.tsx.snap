// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsultTransferListComponent Snapshots Interactions should render component after button click 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper md-focus-ring-wrapper children"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="0"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Interactions should render component after keyboard interaction 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Interactions should render component after list item click 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component should render the component with title and subtitle 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component should render with different button icon 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="different-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component should render with different className 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item different-custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component should render with long name generating different initials 1`] = `
<li
  aria-label="Alexander Benjamin Christopher"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        AB
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      Alexander Benjamin Christopher
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component should render with long subtitle 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Senior Manager of Customer Relations and Business Development
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component should render with single name 1`] = `
<li
  aria-label="John"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        J
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component should render with special characters in title 1`] = `
<li
  aria-label="John O'Connor & Jane"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JO
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John O'Connor & Jane
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component should render without className 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots Rendering - Tests for UI elements and visual states of ConsultTransferListComponent component should render without subtitle 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots State Management should update when buttonIcon changes 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots State Management should update when subtitle changes 1`] = `
<li
  aria-label="John Doe"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JD
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      John Doe
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Senior Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;

exports[`ConsultTransferListComponent Snapshots State Management should update when title changes 1`] = `
<li
  aria-label="Jane Smith"
  class="call-control-list-item custom-class md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="true"
  data-shape="rectangle"
  data-size="50"
  role="listitem"
  tabindex="0"
>
  <div
    class="call-control-list-item-start"
    data-position="start"
  >
    <div
      aria-hidden="false"
      aria-label=""
      class="md-avatar-wrapper"
      data-color="default"
      data-size="32"
      role="img"
    >
      <span
        aria-hidden="true"
        class="md-avatar-wrapper-children"
      >
        JS
      </span>
    </div>
  </div>
  <div
    class="call-control-list-item-middle"
    data-position="middle"
  >
    <mdc-text
      class="md-text-wrapper call-control-list-item-title"
      tagname="div"
      type="body-large-regular"
    >
      Jane Smith
    </mdc-text>
    <mdc-text
      class="md-text-wrapper call-control-list-item-subtitle"
      tagname="div"
      type="body-midsize-regular"
    >
      Manager
    </mdc-text>
  </div>
  <div
    class="call-control-list-item-end"
    data-position="end"
  >
    <div
      class="hover-button"
    >
      <button
        class="md-button-circle-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-ghost="false"
        data-inverted="false"
        data-multiple-children="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="32"
        tabindex="-1"
        type="button"
      >
        <mdc-icon
          name="test-icon"
        />
      </button>
    </div>
  </div>
</li>
`;
